core/src/main/java/com/novoda/noplayer/internal/exoplayer/ExoPlayerFacade.java

```diff
@@ -66,14 +70,69 @@ boolean isPlaying() {
         return exoPlayer != null && exoPlayer.getPlayWhenReady();
     }
 
+    boolean isPlayingAdvert() {
+        return isPlaying() && exoPlayer.isPlayingAd();
+    }
+
+    boolean isPlayingContent() {
+        return isPlaying() && !exoPlayer.isPlayingAd();
+    }
+
+    long mediaDurationInMillis() throws IllegalStateException {
+        assertVideoLoaded();
+        return exoPlayer.getDuration();
+    }
+
     long playheadPositionInMillis() throws IllegalStateException {
         assertVideoLoaded();
         return exoPlayer.getCurrentPosition();
     }
 
-    long mediaDurationInMillis() throws IllegalStateException {
+    long advertBreakDurationInMillis() throws IllegalStateException {
         assertVideoLoaded();
-        return exoPlayer.getDuration();
+        Timeline currentTimeline = exoPlayer.getCurrentTimeline();
+        if (!isPlayingAdvert() || adsLoader.isAbsent() || !(currentTimeline instanceof SinglePeriodAdTimeline)) {
+            return 0;
+        }
+        SinglePeriodAdTimeline adTimeline = (SinglePeriodAdTimeline) currentTimeline;
+        Timeline.Period period = adTimeline.getPeriod(0, new Timeline.Period());
+
+        int currentAdGroupIndex = exoPlayer.getCurrentAdGroupIndex();
+        int advertCount = period.getAdCountInAdGroup(currentAdGroupIndex);
+
+        long advertBreakDurationInMicros = combinedAdvertDurationInGroup(period, advertCount);
+        return C.usToMs(advertBreakDurationInMicros);
+    }
+
+    long positionInAdvertBreakInMillis() throws IllegalStateException {
+        assertVideoLoaded();
+        Timeline currentTimeline = exoPlayer.getCurrentTimeline();
+        if (!isPlayingAdvert() || adsLoader.isAbsent() || !(currentTimeline instanceof SinglePeriodAdTimeline)) {
+            return 0;
+        }
+        SinglePeriodAdTimeline adTimeline = (SinglePeriodAdTimeline) currentTimeline;
+        Timeline.Period period = adTimeline.getPeriod(0, new Timeline.Period());
+
+        int advertCount = exoPlayer.getCurrentAdIndexInAdGroup();
+
+        long playedAdvertBreakDurationInMicros = combinedAdvertDurationInGroup(period, advertCount);
+        return C.usToMs(playedAdvertBreakDurationInMicros) + playheadPositionInMillis();
```

Comment:
If we extract everything after the early return into a function we can invert the if statement to be positive, easier to read than having negations IMHO

Reply:
Same for the previous method

Reply:
4341fb5